rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -3003,6 +3006,37 @@ public void setNeededNodes(int neededNodes) {
         this.monitoring.setNeededNodes(neededNodes);
     }
 
+    @Override
+    public Map<String, List<String>> getInfrasToPoliciesMapping() {
+        Map<String, List<String>> mapping = new HashMap<>();
+        String fileName = null;
+        try {
+            fileName = PAResourceManagerProperties.RM_NODESOURCE_INFRA_POLICY_MAPPING.getValueAsString();
+            if (!(new File(fileName).isAbsolute())) {
+                // file path is relative, so we complete the path with the prefix RM_Home constant
+                fileName = PAResourceManagerProperties.RM_HOME.getValueAsString() + File.separator + fileName;
+            }
+
+            List<String> lines = Files.readAllLines(Paths.get(fileName));
```

Comment:
(Optionnal) I was actually thinking about something alike:
```
Map<String,String[]> validLines = Files.readAllLines(Paths.get(fileName)).stream().map(line -> line.split(",")).filter(line -> (line.length() > 1)).toMap(splittedLine -> splittedLine.get(0).trim(), splittedLine.subList(1, strings.size());
```
(Approximately)
Therefore, you just have a fully prepared structure on which you can iterate to provision your mapping.

I'll leave this suggestion to you own consideration.

Reply:
Good thanks